cpp/src/io/statistics/byte_array_view.cuh

```diff
@@ -0,0 +1,130 @@
+/*
+ * Copyright (c) 2022, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include <cudf/utilities/span.hpp>
+
+namespace cudf::io {
+
+class byte_array_view {
```

Comment:
There might be value in adding a `@brief` description here, indicating that this is like `list_device_view`/`string_view`.
Also, it's a little sad that the changes to `list_device_view` did not pan out.

Reply:
It is very sad, I agree. I actually just pulled this file out into https://github.com/rapidsai/cudf/pull/11322 to ensure that it had visibility. I can add a brief for it, that's a good idea.